### PR TITLE
External Media: Support new Google Photos Picker API

### LINKED ISSUE #40038: External Media: Adapt image selection
Project thread: pcmemI-3sx-p2
Related PR: https://github.com/Automattic/wp-calypso/pull/95880

### PR DESCRIPTION
**Pull request by bogiii**
Fixes https://github.com/Automattic/jetpack/issues/40038

## Proposed changes:

* Hide the new flow behind the feature flag
* Update Google Photos screen supporting new Photos Picker API based on feature flag
  * If the flag doesn't exist, the flow remains the same
* Hide filtering since API doesn't support it anymore; filtering can be done in google's picker screen
Screenshot 2024-11-12 at 18 32 42

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

- Project thread: pcmemI-3sx-p2
- Related PR: https://github.com/Automattic/wp-calypso/pull/95880
- Related PR: https://github.com/Automattic/jetpack/pull/40242

## Testing instructions:

* Setup your sandbox and enable `keyring/includes/services/extended/google-photos.php` -> `is_picker_enabled` by adding your user id
* Go to the post editor, no matter whether simple or atomic site (follow [instructions](https://github.com/Automattic/jetpack/pull/40087#issuecomment-2462035498))
* Add the `/image` block and press "Select image"
* Select `Google Photos` and pick media from the connected account
* Once you are done with the Google Picker, select a media file and press the `Select` button
* Check if the changed flow looks and works fine

### CODE REVIEW

File: projects/plugins/jetpack/extensions/shared/external-media/sources/google-photos/index.js [hunk 1]
```diff
@@ -1,14 +1,41 @@
...
+	if ( ! isAuthenticated ) {
 		return <GooglePhotosAuth { ...props } />;
 	}
 
-	return <GooglePhotosMedia { ...props } />;
+	if ( pickerFeatureEnabled && ! pickerSession?.mediaItemsSet ) {
+		return <GooglePhotosPickerButton { ...props } />;
+	}
+
+	return <GooglePhotosMedia pickerFeatureEnabled { ...props } />;
```
Comment (ouikhuan) [inline projects/plugins/jetpack/extensions/shared/external-media/sources/google-photos/index.js]: Does this means `pickerFeatureEnabled` is true all the time?

**Review by ouikhuan — COMMENTED** [review]
Tested via three different scenario and passed the following using new Google Photos Picker, nice work!
- Simple site ✅ 
- Atomic site ✅ 
- Self-hosted site (JN site) ✅

File: projects/plugins/jetpack/extensions/shared/external-media/sources/google-photos/index.js (hunk 1 above)
Comment (bogiii) [reply projects/plugins/jetpack/extensions/shared/external-media/sources/google-photos/index.js]: Nice catch. I reproduced the issue on the regular flow. 👍

Comment (bogiii) [reply projects/plugins/jetpack/extensions/shared/external-media/sources/google-photos/index.js]: Fixed. Thanks!!!

**Review by ouikhuan — APPROVED** [review]
Thanks for addressing the feedback!

**Pull request merged**